spec/jobs/set_catalog_record_ids_and_barcodes_csv_job_spec.rb

```diff
@@ -65,7 +71,33 @@
 
     it "attempts to update the catalog_record_id/barcode for each druid with correct corresponding catalog_record_id/barcode" do
       expect(bulk_action.druid_count_total).to eq druids.length
-      expect(subject).to have_received(:update_catalog_record_id_and_barcode).with(ItemChangeSet, Hash, buffer).exactly(3).times
+      expect(bulk_action.druid_count_fail).to eq 0
+      expect(subject).to have_received(:update_catalog_record_id_and_barcode).with(ItemChangeSet, Hash, buffer).exactly(druids.length).times
+    end
+  end
+
+  context "with invalid barcode and catalog_record_ids" do
+    let(:csv_file) do
+      [
+        "Druid,Barcode,#{catalog_record_id_column},#{catalog_record_id_column},Refresh",
+        [druids[0], "superbad", catalog_record_ids[0], "#{catalog_record_id_prefix}55555", refresh[0]].join(","),
+        [druids[1], barcodes[1], "trash", "", refresh[1]].join(","),
+        [druids[2], barcodes[2], catalog_record_ids[2], "", refresh[2]].join(",")
+      ].join("\n")
+    end
+
+    describe "#perform" do
+      before do
+        allow(subject).to receive(:with_bulk_action_log).and_yield(buffer)
+        allow(subject).to receive(:update_catalog_record_id_and_barcode)
+        subject.perform(bulk_action.id, {csv_file:})
+      end
+
+      it "only attempts to update the catalog_record_id/barcode for the one druids with valid barcode/catalog_record_id" do
+        expect(bulk_action.druid_count_total).to eq druids.length
+        expect(bulk_action.druid_count_fail).to eq 2
+        expect(subject).to have_received(:update_catalog_record_id_and_barcode).with(ItemChangeSet, Hash, buffer).exactly(druids.length - 2).times
+      end
     end
```

Comment:
adding new tests and extending existing tests so they will continue work regardless of if we have folio vs symphony enabled